text/0683-deprecate-fallback-resolution-ember-resolver.md

```diff
@@ -0,0 +1,139 @@
+---
+Start Date: 2020-11-23
+Relevant Team(s): Ember.js, Learning
+RFC PR: https://github.com/emberjs/rfcs/pull/683
+Tracking: (leave this empty)
+---
+
+# Deprecate Fallback Lookup Paths in ember-resolver
+
+## Summary
+
+Currently we have multiple fallback lookup paths in ember-resolver. As we continue to migrate towards a more strict future where imports are verified at build time (https://github.com/emberjs/rfcs/pull/454) it will be helpful to remove non-strict lookups from ember-resolver, which support many-to-one lookups, in favor of a one-to-one mechanism. This will enable us to make codemods to migrate to:
+Strict Mode (https://github.com/emberjs/rfcs/pull/496)
+the v2 addon format (https://github.com/emberjs/rfcs/pull/507)
+the slimming down the ember-cli build pipeline (https://github.com/emberjs/rfcs/pull/578)
+strict service lookups (https://github.com/emberjs/rfcs/pull/585)
+
+## Motivation
+
+As the community is moving forward with strict lookups having this done in ember-resolver will allow developers to slowly migrate libraries and applications to take advantage of imports verified at build time, the v2 addon format, Strict Mode and Strict Service Lookups. Without ember-resolver support to ensure that applications and libraries use strict lookups, migrating the ecosystem will require application developers using new features to discover issues when integration occurs.
+
+Switching to the strict resolver by default and deprecating the classic resolver opens a migration path for applications that want to ensure compatibility with the new v2 package format. Additionally, apps using the strict resolver will see significant boot-up performance wins, because lookup costs become O(1) instead of O(n).
+
+Ember apps and addons currently default to using https://github.com/ember-cli/ember-resolver to lookup values from the registry. These values include:
+Services
+Components
+Helpers
+Modifiers
+Templates
+ember-strict-resolver defines a stricter and therefore much faster version of these lookups, which results in a 1:1 instead of 1:N resolution.
+
+In the flagship web app at LinkedIn, adopting ember-strict-resolver resulted in the following improvements (per a TracerBench analysis):
+
+boot phase estimated difference -188ms [-195ms to -181ms]
+transition phase estimated difference -42ms [-57ms to -27ms]
+render phase estimated difference -23ms [-37ms to -9ms]
+
+## Detailed design
+
+As libraries migrate to the new use strict format older applications will not be negatively affected as the strict subset resolution is backwards compatible from libraries to applications.
+
+The major difference between the two resolvers can be easily linted against:
```

Comment:
Linted against **how**? Is it limited to confirming that any `injectService` calls for camelcase property names have an argument?